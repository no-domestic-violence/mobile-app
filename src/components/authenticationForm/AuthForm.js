import React, { useState, useRef } from 'react';
import { Text, StyleSheet, Keyboard } from 'react-native';
import {
  StyledInputAuth,
  StyledButton,
  StyledButtonText,
} from '_styles/shared/';
import { Colors } from '_styles/';
import { Divider } from 'react-native-elements';

export default function AuthForm({
  formType,
  headerForm,
  onSubmit,
  buttonText,
}) {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [username, setUsername] = useState('');

  const emailInputRef = useRef();
  const passwordInputRef = useRef();
  const usernameInputRef = useRef();

  const renderUsername = () => {
    return (
      <>
        <StyledInputAuth
          placeholder="Username"
          autoCapitalize="none"
          autoCorrect={false}
          placeholderTextColor="#6c757d"
          onChangeText={setUsername}
          value={username}
          ref={usernameInputRef}
          returnKeyType="next"
          onSubmitEditing={() =>
            emailInputRef.current && emailInputRef.current.focus()
          }
          blurOnSubmit={false}
        />
        <Divider style={{ height: 20, backgroundColor: Colors.primary }} />
      </>
    );
  };

  return (
    <>
      <Text style={styles.header}>{headerForm}</Text>
      {formType === 'sign up' ? renderUsername() : null}
      <StyledInputAuth
        placeholder="Email"
        autoCorrect={false}
        autoCapitalize="none"
        placeholderTextColor="#6c757d"
        onChangeText={setEmail}
        value={email}
        ref={emailInputRef}
        returnKeyType="next"
        onSubmitEditing={() =>
          passwordInputRef.current && passwordInputRef.current.focus()
        }
        blurOnSubmit={false}
      />
      <Divider style={{ height: 20, backgroundColor: '#cadeee' }} />
      <StyledInputAuth
        placeholder="Password"
        autoCapitalize="none"
        autoCorrect={false}
        placeholderTextColor="#6c757d"
        onChangeText={setPassword}
        value={password}
        ref={passwordInputRef}
        returnKeyType="done"
        onSubmitEditing={Keyboard.dismiss}
        blurOnSubmit={false}
        //   secureTextEntry={true}
      />
      <StyledButton onPress={() => onSubmit({ email, password, username })}>
        <StyledButtonText>{buttonText.toUpperCase()}</StyledButtonText>
      </StyledButton>
    </>
  );
}
const styles = StyleSheet.create({
  view: {
    justifyContent: 'flex-end',
    height: '100%',
  },
  header: {
    fontSize: 35,
    fontWeight: '600',
    alignSelf: 'flex-start',
    marginBottom: 40,
    marginLeft: 30,
  }
});
