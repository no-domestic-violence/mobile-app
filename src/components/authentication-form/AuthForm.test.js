import React from 'react';
import { render } from '@testing-library/react-native';
import AuthForm from './AuthForm.component';

const onSubmitForm = jest.fn();
describe('Authentication form component', () => {
  it('should render the right header and username, email, password for sign up', () => {
    const formType = 'sign up';
    const { queryByPlaceholderText, queryByText, queryByTestId } = render(
      <AuthForm
        formType={formType}
        headerForm='Sign up'
        onSubmitForm={onSubmitForm}
        buttonText='Sign up'
      />
    );
    expect(queryByTestId('form-header')).toHaveTextContent('Sign up');
    expect(queryByPlaceholderText(/username/i)).not.toBeNull();
    expect(queryByPlaceholderText(/email/i)).not.toBeNull();
    expect(queryByPlaceholderText(/password/i)).not.toBeNull();
    expect(queryByText('Sign up')).not.toBeNull();
  });

  it('should render the right header and email, password for log in', () => {
    const { queryByPlaceholderText, queryByTestId } = render(
      <AuthForm
        formType='log in'
        headerForm='Log in'
        onSubmitForm={onSubmitForm}
        buttonText='log in'
      />
    );
    expect(queryByTestId('form-header')).toHaveTextContent('Log in');
    expect(queryByPlaceholderText(/email/i)).not.toBeNull();
    expect(queryByPlaceholderText(/password/i)).not.toBeNull();
  });

  it('should render the right header and email, password, old password for change password form', () => {
    const { queryByPlaceholderText, queryByText, queryByTestId } = render(
      <AuthForm
        formType='change password'
        headerForm='Change password'
        onSubmitForm={onSubmitForm}
        buttonText='confirm'
      />
    );
    expect(queryByTestId('form-header')).toHaveTextContent('Change password');
    expect(queryByPlaceholderText(/email/i)).not.toBeNull();
    expect(queryByPlaceholderText(/old password/i)).not.toBeNull();
    expect(queryByPlaceholderText('Password')).not.toBeNull();
    expect(queryByText(/confirm/i)).not.toBeNull();
  });
  it('should match snapshot', () => {
    const result = render(
      <AuthForm
        formType='change password'
        headerForm='Change password'
        onSubmitForm={onSubmitForm}
        buttonText='confirm'
      />
    ).toJSON();
    expect(result).toMatchSnapshot();
  });
});
