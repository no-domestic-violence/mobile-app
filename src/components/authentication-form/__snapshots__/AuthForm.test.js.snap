// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication form component should match snapshot 1`] = `
Array [
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "fontSize": 35,
        "fontWeight": "600",
        "marginBottom": 40,
        "marginLeft": 30,
      }
    }
    testID="form-header"
  >
    Change password
  </Text>,
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCompleteType="off"
        autoCorrect={false}
        blurOnSubmit={false}
        editable={true}
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="#6c757d"
        rejectResponderTermination={true}
        returnKeyType="next"
        secureTextEntry={false}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#fff",
            "borderRadius": 5,
            "color": "#000",
            "flex": 1,
            "fontSize": 18,
            "fontWeight": "400",
            "minHeight": 40,
            "paddingBottom": "4%",
            "paddingLeft": "2%",
            "paddingTop": "4%",
            "width": "85%",
          }
        }
        testID="RNE__Input__text-input"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>,
  <Text
    style={
      Array [
        Object {
          "bottom": 10,
          "color": "red",
        },
      ]
    }
  >
     
  </Text>,
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCompleteType="off"
        autoCorrect={false}
        blurOnSubmit={false}
        editable={true}
        onChangeText={[Function]}
        placeholder="Old password"
        placeholderTextColor="#6c757d"
        rejectResponderTermination={true}
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#fff",
            "borderRadius": 5,
            "color": "#000",
            "flex": 1,
            "fontSize": 18,
            "fontWeight": "400",
            "minHeight": 40,
            "paddingBottom": "4%",
            "paddingLeft": "2%",
            "paddingTop": "4%",
            "width": "85%",
          }
        }
        testID="RNE__Input__text-input"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>,
  <Text
    style={
      Array [
        Object {
          "bottom": 10,
          "color": "red",
        },
      ]
    }
  >
     
  </Text>,
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCompleteType="off"
        autoCorrect={false}
        blurOnSubmit={false}
        editable={true}
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#6c757d"
        rejectResponderTermination={true}
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#fff",
            "borderRadius": 5,
            "color": "#000",
            "flex": 1,
            "fontSize": 18,
            "fontWeight": "400",
            "minHeight": 40,
            "paddingBottom": "4%",
            "paddingLeft": "2%",
            "paddingTop": "4%",
            "width": "85%",
          }
        }
        testID="RNE__Input__text-input"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>,
  <Text
    style={
      Array [
        Object {
          "bottom": 10,
          "color": "red",
        },
      ]
    }
  >
     
  </Text>,
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#415889",
        "borderRadius": 20,
        "fontWeight": "600",
        "marginTop": 40,
        "opacity": 1,
        "overflow": "hidden",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 14,
            "fontWeight": "600",
            "overflow": "hidden",
            "paddingBottom": 10,
            "paddingLeft": 50,
            "paddingRight": 50,
            "paddingTop": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      CONFIRM
    </Text>
  </View>,
]
`;
